lib/borda_poll.rb

```diff
@@ -0,0 +1,46 @@
+require 'set'
+
+class BordaPoll
+  class InvalidCandidateError < StandardError
+  end
+  class MultipleVoteError < StandardError
+  end
+  class OverdueVoteError < StandardError
+  end
+
+  attr_reader :title, :candidates, :votes, :closing
+
+  def initialize(title, candidates, closing)
+    @title = title
+    @candidates = candidates
+    @closing = closing
+    @votes = []
+  end
+
+  def add_vote(vote)
+    unless (vote.candidates.to_set ^ @candidates.to_set).length == 0 then
+      raise InvalidCandidateError.new(vote.candidates)
+    end
+
+    if @votes.map { |element| element.voter }.include?(vote.voter) then
+      raise MultipleVoteError.new(vote)
+    end
+
+    if DateTime.now() > closing then
```

Comment:
- ruby の歴史的な事情なんですが DateTime は deprecated で Time の利用が推奨されています
  - cf. https://docs.ruby-lang.org/ja/latest/class/DateTime.html
  - Time は組み込みクラスなので https://github.com/sankichi92/cookpad-internship-20210218/blob/43f061f546e20b9f47897090df0a5eac03b516e5/spec/borda_poll_spec.rb#L3 も不要になります
- [nit] 引数を渡さないメソッド呼び出しでは `()` を省略できます

```suggestion
    if Time.now > closing
```

Reply:
1つ目に関してはなるほどです。日時を表現するクラスっぽいな～～くらいの気持ちで使っていたんですが非推奨だったのか～～
2つ目は普段書いてる言語の影響で`()`は関数呼び出し演算子(書かないと関数オブジェクトが返ってくる)みたいな気持ちがあったので書いてしまっていました。

Reply:
> 2つ目は普段書いてる言語の影響で()は関数呼び出し演算子(書かないと関数オブジェクトが返ってくる)みたいな気持ちがあったので書いてしまっていました。

わかる（が、もう慣れてしまった）。